### PR TITLE
Support jwt signing

### PR DESCRIPTION
**Pull request by CedrikNikita**
This PR also reduces amount of requests in one minute by `28.58`%.

Requests on heavy loaded wallet in 1 minute.
|Before|After|
|---|---|
|image|image|

### CODE REVIEW

File: src/popup/pages/JwtSign.vue [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      const { callbackOrigin, openCallbackOrGoHome } = useDeepLinkApi();
+      const { nodeNetworkId } = useAeSdk();
+      const { openModal } = useModals();
+      const { setLoaderVisible } = useUi();
+      const { getLastActiveProtocolAccount } = useAccounts();
+
+      try {
+        setLoaderVisible(true);
+        const payload = route.query.payload?.toString();
+        const jwk = route.query.jwk?.toString() === 'true';
```
Comment (thepiwo) [inline src/popup/pages/JwtSign.vue:41]: I wouldn't make this a user choice, lets always include sub_jwk as its a self signed jwt

File: src/popup/pages/JwtSign.vue [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        });
+
+        const dataToSign = JSON.parse(payload);
+
+        if (!jwk) {
+          dataToSign.sub_jwk = undefined;
+        }
+
+        const signature = await signJwt(dataToSign, new AeAccountHdWallet(nodeNetworkId));
+        openCallbackOrGoHome(true, { signature: signature as any });
```
Comment (thepiwo) [inline src/popup/pages/JwtSign.vue:78]: previously it was mentioned by @davidyuk that "signature" is semantically incorrect, as it is includes the payload, so I opted to use "signed-payload" https://github.com/superhero-com/superhero-wallet/pull/2875/files#diff-663c4a09a9d8aef6a1de5c7e9e3b4d84d60e4f9bba55c86b530b2837d8c09d39R58

Comment (thepiwo) [inline src/popup/pages/JwtSign.vue:78]: lets add the address here as parameter

File: src/popup/locales/en.json [hunk 3]
```diff
@@ -307,6 +308,18 @@
         "content": "Transaction data is either broken or Superhero wallet doesn’t support this type of transactions yet. Signing it may cause unexpected results including loss of your funds or other malicious effects.{0}{0}Be careful and sign the data only if you trust the aepp and its owner."
       }
     },
+    "confirm-unsafe-sign": {
+      "heading": "would like you to sign a data.",
+      "superheroChatJwtSign": "would like you to sign a message in order to login.",
+      "title": "Unsafe sign",
+      "typeJwt": "Json Web Token (JWT)",
```
Comment (thepiwo) [inline src/popup/locales/en.json:315]: Json is usually all caps "JSON" as its an abbreviation https://en.wikipedia.org/wiki/JSON

File: src/popup/pages/JwtSign.vue [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+          throw new Error('There is no payload to sign');
+        }
+
+        const payloadAsJson = JSON.parse(payload);
+        if (jwk) {
+          payloadAsJson.sub_jwk = {
+            kty: 'OKP',
+            crv: 'Ed25519',
+            x: toBase64Url(decode(
+              getLastActiveProtocolAccount(PROTOCOLS.aeternity)?.address as Encoded.AccountAddress,
```
Comment (thepiwo) [inline src/popup/pages/JwtSign.vue:54]: its never undefined?

File: src/utils/common.ts [hunk 5]
```diff
@@ -502,3 +502,13 @@ export function getCleanModalOptions<T>(options: T): Omit<T, 'onCompiler' | 'onN
   delete cleanedOptions?.onNode;
   return cleanedOptions;
 }
+
+export const fromBase64Url = (data: string): Buffer => Buffer
+  .from(data.replace(/_/g, '/').replace(/-/g, '+'), 'base64');
+
+export const toBase64Url = (data: Buffer | Uint8Array | string): string => Buffer
```
Comment (thepiwo) [inline src/utils/common.ts:509]: unfortunate it's not exposed by the sdk @davidyuk

File: src/constants/common.ts [hunk 6]
```diff
@@ -422,3 +425,6 @@ export const ASSET_TYPES = {
 export const TRANSACTION_CERTAINLY_MINED_TIME = 600000;
 
 export const POLLING_INTERVAL_TRANSACTIONS = 15000;
+
+// toBase64Url(JSON.stringify({ alg: 'EdDSA', typ: 'JWT' }))
+export const JWT_HEADER = 'eyJhbGciOiJFZERTQSIsInR5cCI6IkpXVCJ9';
```
Comment (thepiwo) [inline src/constants/common.ts:430]: unfortunate it's not exposed by the sdk @davidyuk

File: src/popup/components/Modals/ConfirmUnsafeSign.vue [hunk 7]
```diff
@@ -0,0 +1,208 @@
+<template>
```
Comment (thepiwo) [inline src/popup/components/Modals/ConfirmUnsafeSign.vue:1]: this isn't used when using JWT sign right?

**Review by thepiwo — APPROVED** [review]
it works nicely


with the previous approach to reuse signMessage as path with `?jwt=true` as parameter offered a fallback mechanism for developers to sign something not knowing which wallet version the user uses then either jwt or ae signed message verification

but the new approach is ok I guess

**Comment by Liubov-crypto** [comment]
1. I proposed multisig Tx, approved it by 2nd signer, but the info was not updating for the long time in my wallet: 

https://github.com/superhero-com/superhero-wallet/assets/69896204/39bb52ae-0b35-4d77-b17f-7484d41ba1c3

2. Canceling multisig tx got this error: 

cancel multisg tx


3. For some reason until I revoked my multisig Tx I was not able to send FT, also max button was not working: 

max

https://github.com/superhero-com/superhero-wallet/assets/69896204/b3b80f6a-d4cb-486a-a09c-7621fa016f60

4. minimise wallet and open it again on the latest multisig account: 
https://github.com/superhero-com/superhero-wallet/assets/69896204/8ac034db-e43d-4357-bc35-3f44e8ec3ffb

**Comment by Liubov-crypto** [comment]
LGTM, the last issue is reproducible on dev too, will register it separately.

**Pull request merged**